Clean up many unused declaration/definitions in TH

xuhdev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #29323 Migrate acos from TH to ATen (CUDA)
* **#29046 Clean up many unused declaration/definitions in TH**
* #28953 Migrate frac from TH to ATen (CUDA)

Differential Revision: [D18302767](https://our.internmc.facebook.com/intern/diff/D18302767)

xuhdev committed:
Clean up many unused declaration/definitions in TH

[ghstack-poisoned]

VitalyFedyunin reviewed (approved):
Changes looks good, lets check if all tests are passing

xuhdev committed:
Update on "Clean up many unused declaration/definitions in TH"


Differential Revision: [D18302767](https://our.internmc.facebook.com/intern/diff/D18302767)

[ghstack-poisoned]

xuhdev committed:
Update on "Clean up many unused declaration/definitions in TH"


Differential Revision: [D18302767](https://our.internmc.facebook.com/intern/diff/D18302767)

[ghstack-poisoned]

xuhdev committed:
Update on "Clean up many unused declaration/definitions in TH"


Differential Revision: [D18302767](https://our.internmc.facebook.com/intern/diff/D18302767)

[ghstack-poisoned]
